### PR TITLE
spring mvc tests to java

### PR DESCRIPTION
**Pull request by aschugunov**

### CODE REVIEW

**Comment by steverao** [comment]
There are some CI failure problem in your PR, you can solve them by referring to [relevant document](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/style-guideline.md).

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.core.annotation.Order;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/SecurityConfig.java:18]:
```suggestion
class SecurityConfig {
```

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/ServletFilterConfig.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class ServletFilterConfig {
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/ServletFilterConfig.java:23]:
```suggestion
class ServletFilterConfig {
```

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/ServletFilterTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerTestOptions;
+import io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+
+public class ServletFilterTest extends AbstractServletFilterTest {
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/ServletFilterTest.java:28]:
```suggestion
class ServletFilterTest extends AbstractServletFilterTest {
```

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/ServletFilterTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  static final boolean testLatestDeps = Boolean.getBoolean("testLatestDeps");
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/ServletFilterTest.java:29]: It is better to add `private` keyword and a blank line with above here.

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/ServletFilterTest.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/ServletFilterTest.java:32]:
```suggestion
  private static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
```

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/SpringBootBasedTest.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerTestOptions;
+import io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+
+public class SpringBootBasedTest extends AbstractSpringBootBasedTest {
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/SpringBootBasedTest.java:27]:
```suggestion
class SpringBootBasedTest extends AbstractSpringBootBasedTest {
```

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/SpringBootBasedTest.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/SpringBootBasedTest.java:29]:
```suggestion
  private static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
```

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/SpringBootBasedTest.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+  static final boolean testLatestDeps = Boolean.getBoolean("testLatestDeps");
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/SpringBootBasedTest.java:31]:
```suggestion
  private static final boolean testLatestDeps = Boolean.getBoolean("testLatestDeps");
```

File: instrumentation/spring/spring-webmvc/spring-webmvc-common/testing/src/main/java/boot/SavingAuthenticationProvider.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+package boot;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.authentication.dao.AbstractUserDetailsAuthenticationProvider;
+import org.springframework.security.core.userdetails.UserDetails;
+
+public class SavingAuthenticationProvider extends AbstractUserDetailsAuthenticationProvider {
+  public List<TestUserDetails> latestAuthentications = new ArrayList<>();
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-common/testing/src/main/java/boot/SavingAuthenticationProvider.java:15]:
```suggestion
  List<TestUserDetails> latestAuthentications = new ArrayList<>();
```

File: instrumentation/spring/spring-webmvc/spring-webmvc-common/testing/src/main/java/boot/TestController.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.servlet.view.RedirectView;
+
+@Controller
+public class TestController {
+  @RequestMapping("/basicsecured/endpoint")
```
Comment (steverao) [inline instrumentation/spring/spring-webmvc/spring-webmvc-common/testing/src/main/java/boot/TestController.java:32]: It had better to add a blank line with above here. Other same problems can modify together.

**Comment by aschugunov** [comment]
@steverao thank you a lot for review

**Review by laurit — APPROVED** [review]

**Pull request merged**